Code/autopkglib/Unarchiver.py

```diff
@@ -135,6 +136,12 @@ def main(self):
             ]
         elif fmt == "gzip":
             cmd = ["/usr/bin/ditto", "--noqtn", "-x", archive_path, destination_path]
+        elif fmt == "gunzip":
+            # Since gunzip doesn't decompress to alternate destinations, have to do this later
+            cmd = ["/usr/bin/gunzip", "-dkf", archive_path]
+            # Strip ending '.gz'
+            unarchived_path = archive_path[:-3]
+            mv_cmd = ["/bin/mv", unarchived_path, destination_path]
```

Comment:
My feedback here is still the same: don't shell out to `mv` to move files. Use a native python operation to do this, such as `shutil.move()`:
https://docs.python.org/3/library/shutil.html#shutil.move

This also doesn't incorporate my previous feedback about creating a temporary folder, changing the current directory to it, and gunzipping there.

Reply:
I can use `shutil.move()` to move the decompressed file, but the problem still is even though you might change the current working directory, gunzip will _only_ decompress a .gz file to the same directory as the original file.  As a result, I'm not sure why we'd want to create a temp directory as if we could do this I would think it'd circumvent the issue we're trying to address here in the first place.

Reply:
I think this is to make sure anything unarchived we don't end up using gets flushed/managed by the operating system rather than having to be explicitly purged, since it's a duplication. There's often detritus you'd see alongside source code used to build the artifact like license files and READMEs that can be left behind since we're not going to end up packaging/importing/distributing them.

Reply:
I don't disagree with that.  In my mind that workflow has never been the "standard" autopkg way of doing things though.

By default, Unarchiver takes the archive_file and decompresses it in `%RECIPE_CACHE_DIR%/%NAME%`, or whatever destination_path is specified, meaning that everything - the original downloaded archive file and the decompressed file - are present in the recipe cache.  As such, some recipe authors have acknowledged this by using `PathDeleter` to handle those no longer needed directories, while others leave these folders alone conceivably to not keep what autopkg has done to run the recipe hidden behind the curtain, as it were.  To me, doing what @nmcspadden has suggested means breaking that mold and am just trying to understand why (from a learning / best practice standpoint).